google/cloud/pubsublite/internal/location.h

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_INTERNAL_LOCATION_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_INTERNAL_LOCATION_H
+
+#include "google/cloud/pubsublite/internal/cloud_region.h"
+#include "google/cloud/pubsublite/internal/cloud_zone.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+#include "google/cloud/status_or.h"
+#include "google/cloud/version.h"
+#include "absl/types/variant.h"
+#include <algorithm>
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace pubsublite_internal {
+
+/**
+ * A wrapper around a Google Cloud Location which can be either a `CloudRegion`
+ * or `CloudZone`.
+ */
+class Location {
```

Comment:
This is not in your design doc, and I suspect we won't need it.  Please describe it there and/or remove it from this PR until it is discussed in the design doc.
Ditto for `CloudRegion` and `CloudZone`.

Reply:
These aren't user-facing though? We need these data representations in order to parse region names from the user-provided `Topic` with which we'll create publisher connections.